frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -270,7 +284,27 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
             },
         ],
     })),
-    listeners(({ values, actions, cache }) => ({
+    listeners(({ values, actions, cache, props }) => ({
+        pollRecordingSnapshotsSuccess: () => {
+            // always make sure we've cleared up the last timeout
+            clearTimeout(cache.realTimePollingTimeoutID)
+            cache.realTimePollingTimeoutID = null
+
+            if (values.unnecessaryPollingCount <= 10) {
```

Comment:
Maybe a comment or variable name here would help understand why we picked 10 (and that it can be changed)